security/pkg/caclient/clientrunner.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caclient
+
+import (
+	"fmt"
+	"time"
+
+	"istio.io/istio/pkg/log"
+	pkiutil "istio.io/istio/security/pkg/pki/util"
+	"istio.io/istio/security/pkg/util"
+)
+
+// ClientRunner interacts with the upstream CA to maintain the KeyCertBundle valid.
+type ClientRunner interface {
+	// Run the ClientRunner loop (blocking call)
+	Run(stopCh <-chan struct{}, errCh chan<- error)
```

Comment:
How about adding Stop() method and makes stopCh as struct's field? So that callers don't have to maintain the mapping of stopCh and the runner instance. Will also be consistent with Start/Stop style interface we used in other places.

Reply:
Actually we have a lot of code passing the channels:
https://github.com/istio/istio/search?utf8=%E2%9C%93&q=stop%28%29&type=
Is the Start/Stop style a golang practice? If so, we can use it here.

Reply:
Not a Go thing, but we use this in security/cmd/... , eg. [nodeagent](https://github.com/istio/istio/blob/125fbcf341c570b41ed872a2014880b888fdf139/security/cmd/node_agent_k8s/nodeagentmgmt/nodeagentserver.go#L54)
[workloadhandler](https://github.com/istio/istio/blob/ef8e0b5c53b97b9b11f6b2d892666195163f2de6/security/cmd/node_agent_k8s/workload/handler/workloadhandler.go#L34)

Just like standard library time.Ticker.Stop(), I think calling Stop() is simpler than passing info towards a channel for caller. Also that makes this functionality part of the interface itself.